### DISCUSSION TITLE
Request for CR-10 Smart firmware source

### DISCUSSION THREAD

**Discussion by mitrii** [CrealityOfficial/Creality_Sonic_Pad#20] [General]
I am writing to request the source code for the firmware of the CR-10 Smart product that your company has developed and released under the terms of the GNU General Public License (GPL). As you may be aware, the GPL requires that the source code of any software released under its terms must be made available to anyone who requests it. This is to ensure that users of the software have the freedom to modify, distribute, and use it as they see fit.

I would like to obtain a copy of the source code for the firmware of the CR-10 Smart product so that I can examine it and ensure that your company is in compliance with the GPL. I would appreciate it if you could provide me with the source code as soon as possible.

**Comment by brandonvers** [comment]
hi, did you get the source code?

**Reply by mitrii** [reply]
no, I switched to klipper firmware
